src/datasets/packaged_modules/__init__.py

```diff
@@ -48,6 +50,7 @@ def _hash_python_lines(lines: List[str]) -> str:
     ".parquet": ("parquet", {}),
     ".arrow": ("arrow", {}),
     ".txt": ("text", {}),
+    ".tar": ("webdataset", {}),
```

Comment:
Maybe we can make the module inference more robust by inspecting the contents of TAR archives (e.g., consecutive files with the same name (stem) but different extensions)

Reply:
Some webdatasets may contain only one field (e.g. only images) so I'm not sure it would make sense.

Also I like the idea of keeping the TAR loading simple and only support webdataset for TAR

Reply:
But this would break some existing repos on the Hub, no?

Reply:
TAR has never been supported on the Hub, what would it break ?

Reply:
Ah you mean that some TAR datasets not in webdataset format won't work properly.

Maybe I can add an error message if the first webdataset examples don't have the same type

Reply:
Nevermind, I thought this could fail on https://github.com/huggingface/datasets/blob/27d1fe52857c6a25a29cac63a296405136b2797c/src/datasets/load.py#L449-L450 

But this logic only inspects ZIP (and ignores TAR) archives :)